### PR TITLE
Port/typescript

### PR DESCRIPTION
**Pull request by JonathonRP**
draft to monitor changes

### CODE REVIEW

**Comment by JonathonRP** [comment]
Note this is to just port the current code to typescript,
I have future plans to upgrade code base to align more this latest version of FramerMotion and its new features.

The point of this PR is to allow for more contributions from myself has the code base is hard to follow and noted many dead paths that should be removed but can't be without fear of impacting library functionality

**Comment by JonathonRP** [comment]
@micha-lmxt this could use some testing but this should only change the compile method to support ts and js is valid ts so even if types are not completely correct is should not impede running fine or as before. but I am not able test locally yet to guarantee this

**Comment by reeseperry** [comment]
@micha-lmxt, just tested this and found no issues. should be ready and good to go

**Comment by JonathonRP** [comment]
in testing

**Comment by JonathonRP** [comment]
@micha-lmxt, this is tested now and no immediate issues, seems to function just as before. I will continue playing but this PR seems ready to merge by my account. if any new issues, most likely either pre-existing or were edge cases

**Comment by JonathonRP** [comment]
Fyi there was a change to proxy class that should be mainly style/ formatting change and not functional change
